### PR TITLE
feat: [#504] Route supports configure timeout

### PR DESCRIPTION
**Pull request by KlassnayaAfrodita**
add WithContext

https://github.com/goravel/goravel/issues/504

## 📑 Description

Closes https://github.com/goravel/goravel/issues/504

@coderabbitai summary

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: contracts/http/context.go [hunk 1]
```diff
@@ -25,6 +25,8 @@ type Context interface {
 	Context() context.Context
 	// WithValue add value associated with key in context
 	WithValue(key any, value any)
+
```
Comment (devhaozi) [inline contracts/http/context.go:28]: Pls add comment like other methods.

**Comment by devhaozi** [comment]
And need to fix all tests

File: contracts/http/context.go (hunk 1 above)
Comment (hwbrzzl) [reply contracts/http/context.go:28]: Add sort methods, the `WithContext` should be on top of `WithValue`.

**Comment by KlassnayaAfrodita** [comment]
do I need to implement the withContext function in all files that give an error?

**Comment by kkumar-gcc** [comment]
> do I need to implement the withContext function in all files that give an error?

Yes, you need add `WithContext` to pass type check

**Comment by hwbrzzl** [comment]
CI can't update the mock file automatically if the PR is committed by fork.

**Comment by KlassnayaAfrodita** [comment]
> CI can't update the mock file automatically if the PR is committed by fork.

do i need to generate a mock file and commit it to this pr?

**Comment by hwbrzzl** [comment]
@KlassnayaAfrodita Yes

**Comment by KlassnayaAfrodita** [comment]
I installed mockery and executed '''mockery --all''' in the root of the repository

**Pull request closed without merging**